ui/src/submissions/common/components/SuggesterField.jsx

```diff
@@ -18,19 +20,38 @@ class SuggesterField extends Component {
   }
 
   onChange(value) {
-    const { form, name } = this.props;
+    const { form, name, recordFieldPath } = this.props;
     form.setFieldValue(name, value);
+
+    if (!this.hasChangedBySuggestionSelection) {
+      // flag is necessary to prevent resetting $ref after select for the first time
+      // because after each onSelect, onChange is fired
+      form.setFieldValue(recordFieldPath, null);
+    }
+
+    this.hasChangedBySuggestionSelection = false;
+  }
+
+  onSelect(_, selectedOptionComponent) {
+    const { self } = selectedOptionComponent.props.result._source;
+    const { form, recordFieldPath } = this.props;
+    form.setFieldValue(recordFieldPath, self);
```

Comment:
what will happen if I delete an autocompleted value? Example scenario:

I autocomplete conference record and then delete the value, will the ``conference_record`` field stay?

Reply:
nope, you can see `onChange` where it's set to `null`